n_helpers.c

```diff
@@ -435,39 +514,40 @@ const char * NoteBinaryTransmit(uint8_t * data, size_t dataLen, size_t bufLen,
     // Validate the index provided by the caller, against the `length` value
     // returned from the Notecard to ensure the caller and Notecard agree on
     // how much data is residing on the Notecard.
-    if (offset != len) {
+    if ((long)notecardOffset != len) {
         NOTE_C_LOG_ERROR("notecard data length is misaligned with offset");
         return ERRSTR("notecard data length is misaligned with offset", c_mem);
     }
 
-    // When `offset` is zero, the entire buffer is available
-    const size_t remaining = (offset ? (max - len) : max);
-    if (dataLen > remaining) {
+    // When offset is zero, the entire buffer is available
+    const uint32_t remaining = (notecardOffset ? (max - len) : max);
+    if (unencodedLen > remaining) {
         NOTE_C_LOG_ERROR("buffer size exceeds available memory");
         return ERRSTR("buffer size exceeds available memory", c_mem);
     }
 
     // Calculate MD5
     char hashString[NOTE_MD5_HASH_STRING_SIZE] = {0};
-    NoteMD5HashString(data, dataLen, hashString, NOTE_MD5_HASH_STRING_SIZE);
+    NoteMD5HashString(unencodedData, unencodedLen, hashString, NOTE_MD5_HASH_STRING_SIZE);
 
     // Shift the data to the end of the buffer. Next, we'll encode the data,
     // outputting the encoded data to the front of the buffer.
-    const size_t dataShift = (bufLen - dataLen);
-    memmove(data + dataShift, data, dataLen);
+    const uint32_t dataShift = (bufLen - unencodedLen);
+    memmove(unencodedData + dataShift, unencodedData, unencodedLen);
 
     // outLen holds the buffer size available for encoding. The -1 accounts for
     // one byte of space we need to save for a newline to mark the end of the
     // packet. When NoteBinaryEncode returns, outLen will hold the encoded
     // length.
-    uint32_t encLen = bufLen - 1;
-    err = NoteBinaryEncode(data + dataShift, dataLen, data, &encLen);
+    uint32_t encLen = (bufLen - sizeof(char));
```

Comment:
Truly doesn't matter, but why use parentheses here?

Reply:
it is a habit I have for anytime there is a compound expression.

If the expression were to grow, then you must consider the parenthesis instead of just carelessly appending.

For example, imagine you wanted to double the size.

:point_down: oops!
```c
uint32_t encLen = bufLen - sizeof(char) * 2;
```
```c
uint32_t encLen = (bufLen - sizeof(char)) * 2;
```
:point_up_2: :+1: